### PR TITLE
GTEST/UCS: Fix prints in test_async

### LINKED ISSUE #4251: Question about UCP version message during "make gtest"
I see this while testing the current (and some previous) git HEAD

```shell
[     INFO ] < UCP version is incompatible, required: 99.99, actual: 1.8 (release 0 /home/tc/ucx-gtest/source/src/ucp/.libs/libucp.so.0) >
```

Anything to worry about?

### PR DESCRIPTION
**Pull request by brminich**
## What
Fixes prints in test_async.warn_block

## Why ?
Fixes #4251 
No need to print all leftover warnings from previous tests

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12511/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 4 of 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 11-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/d82c07161ff478b6ef459c5255e0d2761cb42aeb/_main) | :x: FAILURE
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/d28ba82fc874b859b7a702a4681acfc775bc5176/hpc-test-node-gpu_W1) | :x: FAILURE
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/b9a7151d626566c354caabe82f56a00a87cc3763/hpc-test-node-legacy_W1) | :x: FAILURE
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/a0294b3bd2acf99693612a3218c8e2ce8d79234b/hpc-test-node-new_W1) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/79e5c3ebdb162686bdc37cb9962e203685364db3/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/9ba6081b9a6ce19f3a0755a11f0642e60d02a51b/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/514575c5ba73795525eadcb42060075e1b177eb3/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/c0e40d213e32a775b143df653f40d4b03ab338ad/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/ee1deb311f4e33e058977fa282ad5d09d7b8e762/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/4add77f4d50788846a371a0373ce001b192ea870/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/51ade3aca93f61ae5fd6925b844474a091a895fa/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/8a7dd10eed881833312c07b96e2dcca668917552/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/cf169cca2d9cbb38ec4fec42486178445deff91c/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/d5a68c834f380f9cb4901cb125e822b7202667f3/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/2318bf6a1423843a31abde5efbbc900d2ed8b517/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/587f4148ad89d8d0026fdd000d49060a5b229554/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/868efb84771d4da2167b77e22661852e62922093/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/882fb3072115cd7050704cb699906a4fba398957/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/4dc73c27cb31d85404d1050dd323fa0648ccca94/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/0b6c70174efe08a1a75a2dd4199861ab4ddc57c6/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/7369c7a71b6800b97e4b0c8a7078a3de58878da7/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/462f234b128ef1e6d7efe75177ab8be68ba29d2b/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/27367488db424087f2b7f177ef4ee9edd51d0284/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/aab7d94044057d552d611466b90456eaec1ca870/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/0d1ab8ecc3589e30f176b308e38153197ee674db/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/93dcfd3d58a10c097317e23e57f6c57b13e39a05/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/7233777d40ae93339f3d9ffc59200867ec443208/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/d31e73f149ac650d31c5e6f817907213e28aa72a/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/8d972bd590cc4ecac92a73e2ef2c0d54/raw/359dfdbfe1947172c9ffd5c801ab010be255d0b2/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12513/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 12-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/66d88a1188fb228f9a25e4e36baf4a997279d792/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/9d6c34250c632d57ca24e97f8d5c4663a1b0c6ef/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/f94bc6c9a1ae1b0c8902285c3a02c448128a2a47/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/d2c7792488e25cd33f4ea5098cf5e11fa2264122/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/4d4dc8304973a63359c70136508dc50be0013b83/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/064fb970fb46a2f1f40b68bcd9709aef703daa36/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/eb0d1cdb5cc22460bf3bea980bb3e5503d24b613/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/c7a8b9b12e324a6890e6c1ce5f0478a8b931ce5e/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/30bd2bb96106dc6f1ab40bfe5020542b604fa535/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/a42ffc26eac11e71f73b6e847a3cdca0f53ca569/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/07c3264df4a8c7910a697e04ee209f34a0f3f487/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/fb62ab0cbb4ab5ce9ccdc893b01081e8504b329f/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/82e5d7fb21518de61531e44c43b8073e80f9b7fe/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/3f57e3570649c93036c65296681624a2a6df239e/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/22d7969bfc44d767ff88c465c549364d1464f6f4/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/9bd03b8b70661296cef0e42dceabaa20b044b441/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/b5d2a853bf1fcf0b468eb0b45f74fde1aaa7f3fd/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/06cb798c4c1d24d97f2db7d079ef4bbcf91da6ed/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/a97a71af6c9b837b88e22e67946ae9a3b447d3ad/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/0ca0f7c231f142adc53fcc5fdb16127981764312/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/fd341dd265f33659ce94efeb9591cd07f80e4700/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/69f6c0ada85f1986226d1ee9d9a5328166c9e5fd/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/d6c5c2a0a1fcf03ef847ec93d750dfb404a34d62/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/1903f0f3302342c3e6994b875db44a49c541e12d/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/25cb06f182a0fdc4d2f390d7d6a5775de4efaa66/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/e9b9e10511a6d02c8f4e110b544f0ffdba89ad5d/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/826a02e57effda52431aab4e07ebb698939859da/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/21a26707584a91821cb597c5b67ce62ecb12ba9f/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/7fd977850051c68c34c71b3045f22311/raw/2116dbf4ff2769b08d110f428835fff81ea8056a/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by brminich** [comment]
@hoopoepg, plz check as well

File: test/gtest/ucs/test_async.cc [hunk 1]
```diff
@@ -595,13 +596,11 @@ UCS_TEST_P(test_async, warn_block) {
...
 
-    int warn_count = m_warnings.size();
-    for (int i = 0; i < warn_count; ++i) {
-        UCS_TEST_MESSAGE << "< " << m_warnings[i] << " >";
-    }
-
     if (GetParam() != UCS_ASYNC_MODE_POLL) {
-        EXPECT_GE(warn_count, 1);
+        if (!m_warnings.empty()) {
+            UCS_TEST_MESSAGE << "< " << m_warnings.back() << " >";
```
Comment (yosefe) [inline test/gtest/ucs/test_async.cc]: @brminich instead, need to clear warnings before each test in https://github.com/openucx/ucx/blob/master/test/gtest/common/test.cc#L222

Comment (brminich) [reply test/gtest/ucs/test_async.cc]: @yosefe, #4282

**Pull request merged**